src/torch_fourier_slice/slice_extraction/_extract_central_slices_rfft_2d.py

```diff
@@ -0,0 +1,83 @@
+import einops
+import torch
+from torch_image_lerp import sample_image_2d
+
+from ..dft_utils import fftfreq_to_dft_coordinates
+from ..grids.central_line_fftfreq_grid import central_line_fftfreq_grid
+
+
+def extract_central_slices_rfft_2d(
+    image_rfft: torch.Tensor,
+    image_shape: tuple[int, int],
+    rotation_matrices: torch.Tensor,  # (..., 2, 2)
+    fftfreq_max: float | None = None,
+) -> torch.Tensor:
+    """Extract central slice from an fftshifted rfft."""
+    # generate grid of DFT sample frequencies for a central slice spanning the x-plane
+    freq_grid = central_line_fftfreq_grid(
+        image_shape=image_shape,
+        rfft=True,
+        fftshift=True,
+        device=image_rfft.device,
+    )  # (w, 2) yx coords
+
+    # keep track of some shapes
+    stack_shape = tuple(rotation_matrices.shape[:-2])
+    rfft_shape = (freq_grid.shape[-2],)
+    output_shape = (*stack_shape, *rfft_shape)
+
+    # get (b, 2, 1) array of yx coordinates to rotate
+    if fftfreq_max is not None:
+        normed_grid = einops.reduce(freq_grid**2, "w yx -> w", reduction="sum") ** 0.5
+        freq_grid_mask = normed_grid <= fftfreq_max
+        valid_coords = freq_grid[freq_grid_mask, ...]  # (b, yx)
+    else:
+        valid_coords = einops.rearrange(freq_grid, "w yx -> (w) yx")
```

Comment:
this rearrange looks useless, just leftover from copying the 3d -> 2d impl. where it did something?

Reply:
you're right! I misunderstood what this einops command did. Also the normed grid was useless for the 1D case :+1: